src/builder-context.h

```diff
@@ -22,14 +22,22 @@
 #define __BUILDER_CONTEXT_H__
 
 #include <gio/gio.h>
-#include <libsoup/soup.h>
 #include <curl/curl.h>
 #include "builder-options.h"
 #include "builder-utils.h"
 #include "builder-sdk-config.h"
 
 G_BEGIN_DECLS
 
+/* Same as SOUP_HTTP_URI_FLAGS, means all possible flags for http uris */
+
+#if GLIB_CHECK_VERSION (2, 68, 0) || !GLIB_CHECK_VERSION (2, 66, 0)
+#define CONTEXT_HTTP_URI_FLAGS (G_URI_FLAGS_HAS_PASSWORD | G_URI_FLAGS_ENCODED_PATH | G_URI_FLAGS_ENCODED_QUERY | G_URI_FLAGS_ENCODED_FRAGMENT | G_URI_FLAGS_SCHEME_NORMALIZE)
+#else
+/* GLib 2.66 didn't support scheme-based normalization */
```

Comment:
We don't actually care unless we are doing comparisons between URIs. (We may or may not, I haven't looked).

Reply:
Sure, this was mostly copy-paste from flatpak which did similar migration earlier. Afaik the thought behind is that with at least newer versions of glib you get behaviour that is equivalent to the old behaviour.